File: Week_10/Practicum/Find Champion II.py
class Solution:
    def findChampion(self, n: int, edges: List[List[int]]) -> int:
        in_count = [0 for _ in range(n)]

        for edge in edges:
            in_count[edge[1]] += 1

        champions = [k for k, v in enumerate(in_count) if v == 0]

        return champions[0] if len(champions) == 1 else -1

File: Week_10/Practicum/Find Champion II.cpp
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        vector<vector<int>> graph(n);
        vector<int> indegree(n, 0);

        for (auto &e : edges) {
            graph[e[0]].push_back(e[1]);
            indegree[e[1]]++;
        }

        int champion = -1;
        
        for (int i = 0; i < n; i++) {
            if (indegree[i] == 0) {
                if (champion == -1) {
                    champion = i; 
                } else {
                    return -1;
                }
            }
        }

        return champion;
    }
};
